### PR TITLE
feat: use axe rule tags to provide WCAG information

### PR DESCRIPTION
**Pull request by dbjorge**
#### Description of changes

This replaces the old behavior of providing WCAG info based on hard-coded rule-id-to-wcag-standards data to using axe rule tags to determine WCAG info. It still hard codes the links to the WCAG standards based on the axe tags, since we don't currently have a way to infer the actual links from axe-core results alone.

#### Pull request checklist

- [x] Addresses an existing issue: Implements user story #1505355
- [x] Added relevant unit test for your changes. (`npm run test`)
- [x] Verified code coverage for the changes made.

### CODE REVIEW

File: src/result-decorator.test.ts [hunk 1]
```diff
@@ -23,60 +17,97 @@ describe('Result Decorator', () => {
...
     });
 
-    it('result decorator contains WCAG information that is provided as dependency', () => {
+    // tslint:disable-next-line: mocha-no-side-effect-code
+    const resultsArrayProperties = [
+        'passes',
+        'violations',
+        'inapplicable',
+        'incomplete',
+    ] as const;
```
Comment (Shobhit1) [inline src/result-decorator.test.ts]: What does as `const` do? I am a little confused by this.

Comment (peterdur) [reply src/result-decorator.test.ts]: Const assertions tell TypeScript you're intending a deeper level of immutability than just the reference. [More details on TypeScript site](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions).

Comment (dbjorge) [reply src/result-decorator.test.ts]: This is a typescript 3.4 feature called a "const assertion". You can read full details in the [typescript 3.4 release notes](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html).

In this particular case, using it turns the type of resultsArrayProperties from `string[]` to the constant tuple type `readonly ["passes", "violations", "inapplicable", "incomplete"]`. This is useful here because it lets us avoid casting later; specifically, it enables lines 53 and 75 to understand that the type of `resultStub[resultsArrayPropertyUnderTest]` is going to specifically match the type of one of `resultsStub.passes`, `resultsStub.violations`, etc, rather than the weaker understanding that it will match the type of *some arbitrary* property of `resultsStub` (which is all the type checker could understand if it thought `resultsArrayPropertyUnderTest` was an arbitrary `string` as opposed to one of the 4 known constant elements of our constant tuple type).

File: src/wcag-link-data.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    return {
+        text: `WCAG ${sectionNumber}`,
+        title: title,
+        url: `https://www.w3.org/WAI/WCAG21/Understanding/${urlFormattedTitle}`,
+    };
+}
+
+// prettier-ignore
+export const axeTagsToWcagLinkData: DictionaryStringTo<WCAGLinkData> = {
+    wcag111: standardWcag21Link("non-text-content", '1.1.1', 'Non-text Content'),
```
Comment (peterdur) [inline src/wcag-link-data.ts]: Is it intentional to have double quotes for `urlFormattedTitle` but single quotes for the other strings?

File: src/result-decorator.test.ts [hunk 3]
```diff
@@ -23,60 +17,97 @@ describe('Result Decorator', () => {
...
+
+    // tslint:disable-next-line: mocha-no-side-effect-code
```
Comment (dbjorge) [inline src/result-decorator.test.ts]: I didn't want to sneak a tslint update into this same PR, but I'm on the fence about disabling this rule entirely; it's a valuable check (it prevents a class of bug we talked about in the last engineering discussion), but it's overzealous in that it also precludes the use `.each`, which isn't great.

File: src/wcag-link-data.ts (hunk 2 above)
Comment (peterdur) [inline src/wcag-link-data.ts]: Not blocking: I wonder if this would be easier to read/review in the order `sectionNumber`, `title`, `urlFormattedTitle` since then the things that should match (key and `sectionNumber`, `title` and `urlFormattedTitle`) would be adjacent. Actually... on a quick skim it seems like we could programmatically generate `urlFormattedTitle` from `title` (lowercase, remove parens, replace spaces with hyphen), any interest in doing that?

Comment (dbjorge) [reply src/wcag-link-data.ts]: I agree on the ordering (fixed now). I tried inferring the urlFormattedTitle and remember finding some edge case where it was ambiguous whether to include a source `-` or not, so I abandoned that.

File: src/result-decorator.test.ts (hunk 3 above)
Comment (dbjorge) [reply src/result-decorator.test.ts]: Peter suggested a way of doing it that was very reasonable/non-contentious, so I've included that.

File: src/wcag-link-data.ts (hunk 2 above)
Comment (dbjorge) [reply src/wcag-link-data.ts]: No, fixed

File: src/result-decorator.test.ts (hunk 1 above)
Comment (Shobhit1) [reply src/result-decorator.test.ts]: Thanks guys. Appreciate the links 👍
